core/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java

```diff
@@ -314,4 +316,76 @@ public void testEmptyName() throws IOException {
             assertThat(e.getMessage(), containsString("name cannot be empty string"));
         }
     }
+
+    public void testOutOfRangeValues() throws IOException {
+        final List<Triple<String, Object, String>> inputs = Arrays.asList(
+            Triple.of("byte", "128", "is out of range for a byte"),
+            Triple.of("short", "32768", "is out of range for a short"),
+            Triple.of("integer", "2147483648", "For input string"),
+            Triple.of("long", "92233720368547758080", "For input string"),
+
+            Triple.of("half_float", "65504.1", "[half_float] supports only finite values"),
+            Triple.of("float", "3.4028235E39", "[float] supports only finite values"),
+            Triple.of("double", "1.7976931348623157E309", "[double] supports only finite values"),
+
+            Triple.of("half_float", Float.NaN, "[half_float] supports only finite values"),
+            Triple.of("float", Float.NaN, "[float] supports only finite values"),
+            Triple.of("double", Double.NaN, "[double] supports only finite values"),
+
+            Triple.of("half_float", Float.POSITIVE_INFINITY, "[half_float] supports only finite values"),
+            Triple.of("float", Float.POSITIVE_INFINITY, "[float] supports only finite values"),
+            Triple.of("double", Double.POSITIVE_INFINITY, "[double] supports only finite values")
+        );
+
+        for(Triple<String, Object, String> item: inputs) {
+            try {
+                parseRequest(item.type, createIndexRequest(item.value));
+                fail("Mapper parsing exception expected for [" + item.type + "] with value [" + item.value + "]");
+            } catch (MapperParsingException e) {
+                assertThat("Incorrect error message for [" + item.type + "] with value [" + item.value + "]",
+                    e.getCause().getMessage(), containsString(item.message));
+            }
+        }
+    }
+
+    private static class Triple<K,V,M> {
```

Comment:
Instead of this being a generic class with a generic name can we call this `OutOfRangeSpec`, remove the generic arguments and instead use `K -> String`, `V -> Number`, `M -> String`. Also I think it would be ok for you to declare this class here and then reuse it in the `NumberFieldTypeTests` below instead of re-defining it